src/Security/Authorization/Core/src/AuthorizationPolicyCache.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Authorization;
+
+/// <summary>
+/// Represents a cache for an AuthorizationPolicy instance
+/// </summary>
+public class AuthorizationPolicyCache
+{
+    /// <summary>
+    /// The cached policy.
+    /// </summary>
+    public AuthorizationPolicy? Policy { get; set; }
```

Comment:
Also should this metadata be added for MVC as well? Maybe in [here ](https://github.com/dotnet/aspnetcore/blob/3ea008c80d5cc63de7f90ddfd6823b7b006251ff/src/Mvc/Mvc.Core/src/ApplicationModels/AuthorizationApplicationModelProvider.cs#L11).

Reply:
I'm not sure, this is mostly a perf optimization, I'm going to leave it out of this PR for now, but it is a good idea and I will file a tracking issue for considering this for MVC